File: test/fixtures/rust/post_with_quotes_anywhere.rs
extern crate reqwest;
use reqwest::header;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut headers = header::HeaderMap::new();
    headers.insert("A", "''a'".parse().unwrap());
    headers.insert("B", "\"".parse().unwrap());
    headers.insert(header::COOKIE, "x=1'; y=2\"".parse().unwrap());
    headers.insert("Content-Type", "application/x-www-form-urlencoded".parse().unwrap());

    let client = reqwest::blocking::Client::builder()
        .redirect(reqwest::redirect::Policy::none())
        .build()
        .unwrap();
    let res = client.post("http://localhost:28139")
        .basic_auth("ol'", Some("asd\""))
        .headers(headers)
        .body("a=b&c=\"&d='")
        .send()?
        .text()?;
    println!("{}", res);

    Ok(())
}

File: test/fixtures/r/post_with_quotes_anywhere.r
library(httr)

cookies = c(
  x = "1'",
  y = '2"'
)

headers = c(
  A = "''a'",
  B = '"',
  `Content-Type` = "application/x-www-form-urlencoded"
)

data = "a=b&c=\"&d='"

res <- httr::POST(url = "http://localhost:28139", httr::add_headers(.headers=headers), httr::set_cookies(.cookies = cookies), body = data, httr::authenticate("ol'", 'asd"'))
